### PR TITLE
22672 - Fix missing Approved invoices on get_invoices_owing_amount

### PR DESCRIPTION
**Pull request by Jxio**
*Issue #:*
https://github.com/bcgov/entity/issues/22672

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

**Comment by ochiu** [comment]
Thanks Jia, can we add / expand on an existing test to check on this invoices amount owing please?

File: pay-api/src/pay_api/services/statement.py [hunk 1]
```diff
@@ -248,6 +248,7 @@ def get_invoices_owing_amount(auth_account_id: str):
                 .filter(InvoiceModel.invoice_status_code.in_((InvoiceStatus.SETTLEMENT_SCHEDULED.value,
                                                               InvoiceStatus.PARTIAL.value,
                                                               InvoiceStatus.CREATED.value,
```
Comment (seeker25) [inline pay-api/src/pay_api/services/statement.py:250]: Take out CREATED, PARTIAL

Also shouldn't we filter for EFT here?

File: pay-api/src/pay_api/services/statement.py [hunk 2]
```diff
@@ -248,6 +248,7 @@ def get_invoices_owing_amount(auth_account_id: str):
                 .filter(InvoiceModel.invoice_status_code.in_((InvoiceStatus.SETTLEMENT_SCHEDULED.value,
```
Comment (seeker25) [inline pay-api/src/pay_api/services/statement.py:248]: SETTLEMENT_SCHED wont happen for EFT

Comment (seeker25) [inline pay-api/src/pay_api/services/statement.py:248]: Do we need an EFT filter here?

File: pay-api/src/pay_api/services/statement.py (hunk 1 above)
Comment (seeker25) [inline pay-api/src/pay_api/services/statement.py:250]: I think we need to take out CREATED

Comment (seeker25) [reply pay-api/src/pay_api/services/statement.py:250]: The parent function doesn't have CREATED:

image

File: pay-api/src/pay_api/services/statement.py (hunk 2 above)
Comment (seeker25) [reply pay-api/src/pay_api/services/statement.py:248]: I think we do

Comment (ochiu) [reply pay-api/src/pay_api/services/statement.py:248]: yes, let's do that to make this more clean/safe. Only EFT uses the statement summary right now. We can consider expanding this by adding a payment method query parameter in the future if needed.

**Comment by seeker25** [comment]
Associated this to the correct ticket 22672

**Review by seeker25 — APPROVED** [review]

**Pull request merged**